Add :ttl configuration option

pkarman opened the pull request:
**Why**: The :expire_after config option controls both the Redis
session expiration and the expiration of the Rails session cookie.
Adding a new config option :ttl allows those controls to be
managed separately.

The most common case would be if you wanted the Rails cookie
to expire when the browser is closed, as it would if :expire_after
were set to `nil`.

jessieay commented on spec/redis_session_store_spec.rb:
```diff
@@ -59,6 +59,28 @@
+    end
+
+    it 'assigns the :ttl option to @default_options' do
+      expect(default_options[:ttl]).to eq(60 * 120)
```
maybe we can assign `60 * 120` to a variable and ref that here and on L72 so it's more obvious where it's coming from?

jessieay commented on lib/redis-session-store.rb:
```diff
@@ -128,6 +128,11 @@ def set_session(env, sid, session_data, options = nil)
   alias write_session set_session
 
+  def get_expiry(env, options)
+    opts = options || env.fetch(ENV_SESSION_OPTIONS_KEY)
```
what about calling this `expiry` instead of `opts` ? That seems clear-er to me.

pkarman replied:
it's not really `expiry` though -- its all the session storage options.

jessieay replied:
oh you are right, I was misreading what was up there before. My basic feeling on this is that having `options` and `opts` vars in the same method is tricky. If we can find slightly better names for one or both, that would be 🆒

pkarman replied:
how about `redis_options` instead of `opts` ?

pkarman replied:
or `session_storage_options` ?

jessieay replied:
sure! both work for me

pkarman commented:
`session_storage_options` it is @jessieay

jessieay reviewed (approved):
🐶

monfresh commented on lib/redis-session-store.rb:
```diff
@@ -128,6 +128,11 @@ def set_session(env, sid, session_data, options = nil)
   alias write_session set_session
 
+  def get_expiry(env, options)
+    session_storage_options = options || env.fetch(ENV_SESSION_OPTIONS_KEY)
```
Will `ENV_SESSION_OPTIONS_KEY` always be present? If not, then perhaps use `fetch` with a default value, such as `env.fetch(ENV_SESSION_OPTIONS_KEY, {})`? That way, if session_storage_options is nil, `session_storage_options[:ttl]` won't raise a NoMethodError.

pkarman replied:
The existing code seems to assume that `env.fetch(ENV_SESSION_OPTIONS_KEY)` always exists, since I just cut/pasted that line. Happy to be a little defensive though.

zachmargolis commented:
Is it worth updating the README as well?

pkarman commented:
@zachmargolis README updated.

pkarman commented:
@monfresh @zachmargolis comments addressed, thanks. PTAL

zachmargolis commented on README.md:
```diff
@@ -36,7 +36,8 @@ In your Rails app, throw in an initializer with the following contents:
   redis: {
-    expire_after: 120.minutes,
+    expire_after: 120.minutes,  # cookie expiration
+    ttl: 120.minutes,           # Redis expiration
```
"redis expiratoin, defaults to `expire_after`" ?

zachmargolis reviewed (approved):
LGTM

Commit:
Add :ttl configuration option

**Why**: The :expire_after config option controls both the Redis
session expiration and the expiration of the Rails session cookie.
Adding a new config option :ttl allows those controls to be
managed separately.

The most common case would be if you wanted the Rails cookie
to expire when the browser is closed, as it would if :expire_after
were set to `nil`.
